### PR TITLE
NXPY-74: Support Operation context

### PR DESCRIPTION
**Pull request by ReneVolution**
I have created this simple fix to support `context` in an Operation payload (as required by e.g. `FileManager.Import`). Following sample code works now as expected:

```
    nx = get_nuxeo_client()
    operation = nx.operations.new('FileManager.Import')
    operation.params = {
        'context': {
            'currentDocument': "/default-domain/UserWorkspaces/Administrator"
        }
    }
    operation.input_obj = blob
    operation.execute()
```

### CODE REVIEW

**Comment by LKleinNux** [comment]
From what I gather from the documentation, the context is not a parameter of the operation. It seems inconsistent to add it to the `operation.params` dictionary, only to get it back out while executing the operations.py::`API.get_params()` method.
It would make more sense to add it as a property of the models.py::`Operation` class, and be able to do `operation.context = {...}`. To include it in the request, you would then check if it has a value in the operations.py::`API.execute()` and add it to `data`, just like we do already for the `input_obj`.

**Comment by ReneVolution** [comment]
I agree 💯and I saw that too - just had not the time to do it that way. I may find some soon and improve the PR.

**Comment by ReneVolution** [comment]
@LKleinNux I have adjusted the PR slightly for a more proper implementation. Let me know what you think.

File: nuxeo/operations.py [hunk 1]
```diff
@@ -221,33 +221,46 @@ def get_attributes(operation, **kwargs):
...
             params = operation.params
         else:
             command = kwargs.pop('command', None)
             input_obj = kwargs.pop('input_obj', None)
+            context = kwargs.pop('context', None)
             params = kwargs.pop('params', kwargs)
-        return command, input_obj, params
+        return command, input_obj, params, context
+
+    def build_payload(self, params, context):
```
Comment (BoboTiG) [inline nuxeo/operations.py]: Could you add a docstring and types comment please?

Comment (BoboTiG) [inline nuxeo/operations.py]: Could you add a docstring and types comment please?

Comment (ReneVolution) [reply nuxeo/operations.py]: @BoboTiG Just added both to the PR.

File: tests/test_operation.py [hunk 2]
```diff
@@ -58,3 +58,10 @@ def test_params_setter(server):
     assert params['param1'] == 'bar'
     assert params['param2'] == 'bar'
     assert params['param3'] == 'plop'
+
+
+def test_context_setter(server):
```
Comment (BoboTiG) [inline tests/test_operation.py]: This is almost good :)
Could you update to use this code:
```diff
diff --git a/tests/test_operation.py b/tests/test_operation.py
index 9c6d806..e614cf8 100644
--- a/tests/test_operation.py
+++ b/tests/test_operation.py
@@ -63,5 +63,6 @@ def test_params_setter(server):
 def test_context_setter(server):
     operation = server.operations.new('Noop')
     operation.context = {'currentDocument': 'foo'}
-    context = operation.context
-    assert context['currentDocument'] == 'foo'
+    with pytest.raises(HTTPError):
+        operation.execute()
+    assert operation.context['currentDocument'] == 'foo'
```
Like that the `build_payload()` method will be entirely covered.

Comment (BoboTiG) [reply tests/test_operation.py]: Or even better, I would be happier to merge a complete test reflecting your use case (using `FileManager.Import` for example, reduced to the minimum). Do you think it is possible?

Comment (ReneVolution) [reply tests/test_operation.py]: I am going to apply the simple change. Not sure when I find time for the additional test.

**Pull request merged**